### DISCUSSION TITLE
3.7.3 to 3.8.9 ,  High memory usage about quorum_queue_procs (20G, 80%!!!)

### DISCUSSION THREAD

**Discussion by feiniao886** [rabbitmq/rabbitmq-server#2863] [Other]
3.7.3 to 3.8.9 ,  High memory usage about quorum_queue_procs (20G, 80%!!!)

but  queue type: Classic, not Quorum

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
This is a bit light on details. We cannot suggest much. Please post the output of `rabbitmq-diagnostics` as explained in [Reasoning About Memory Use](https://www.rabbitmq.com/memory-use.html).

[Quorum queue resource use](https://www.rabbitmq.com/quorum-queues.html#resource-use) depends on two factors: Raft WAL segment file size and how large a queue's in memory log table is allowed to grow.

You may have a lot of quorum queues in a different virtual host. Classic queue memory footprint cannot be counted in the quorum queue category.

**Reply by michaelklishin** [reply]
I should have mentioned that if you use a lot of quorum queues, you should adjust their max allowed resource use. Defaults assume a small number of busy QQs. For example, WAL segment file size can be reduced to something like 16-32 megabytes.
In-memory log table size (which is per queue) can be in the 8-16 megabyte range. Then see if the disk I/O usage is appropriate for your needs, if it is, those values are fine to use.

**Reply by feiniao886** [reply]
Thank you for your prompt reply.
All queue types are classic. The arbitration queue type is not used.
This is my question
thx

**Reply by feiniao886** [reply]
![image](https://user-images.githubusercontent.com/29135686/109895530-55d20f80-7cca-11eb-97b0-5cedd497d82f.png)

**Reply by michaelklishin** [reply]
Arbitration?

MQTT plugin tracks client IDs using Raft, which is currently classified as "quorum queues".

Starting with 3.8.10, the [deletes of tracked IDs is a lot more efficient](https://github.com/rabbitmq/rabbitmq-server/pull/2692). Please upgrade to the [latest release](https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.8.14).
